### ISSUE TITLE
Code Review Week 3

### ISSUE THREAD

**Issue by koshka** [Oluwatosin12-bit/Kudos-Board#1]
Hi @Oluwatosin12-bit! Good job finishing your third project 👏

Here's my feedback for this week's work:

### Version control / Github
I didn't see you using Pull Requests (PRs) this week. I believe this is a required part of the workflow.
Let's go into details during our 1:1, but briefly the expected process is:
- each feature of a project is worked on in a separate branch
- once the feature is ready, the branch is merged into `main` via a PR and can be deleted
- PRs should be relatively small, so they are easy to review
- each PR is reviewed by your IM and peers
Starting on week 4 your should merge PRs daily and the code reviews are also going to happen daily, so you can get more frequent feedback. This is actually closer to the real-world development cycle, where code is reviewed frequently and in small chunks 🙂

Here's more info about PRs from the intern guide: https://sites.google.com/meta.com/metau-intern/code-guidance/pullrequests
It also mentions this template that you should add to your repo, so it's easier to write PR descriptions: https://docs.google.com/document/d/1XWjwOWVYDaEj-31BMOzZMoRyr-O01sS4HpB-PtgGbuc/edit 

### Features / UX
* If you open a board and look at Network tab in the developer tools, looks like the code is constantly sending GET requests for the cards. You have an infinite loop somewhere
* When I open a board and refresh the page, I get "Not Found" error
* I noticed that the footer is jumping up when there isn't much content on the page. It would be better if it stayed at the bottom of the page, regardless of how much content is there
* if you go to one of the categories, select "View board" and then press "Back", selected category gets cleared; Ss this the intended behavior? As a user I'd expect that the selection stayed until cleared
* is this expected that images on the boards are updated after page refresh?

Some nits ("nit picking"). Just my personal preference, not bugs
* currently selected category isn't highlighted; when a category button is pressed, it turns purple, but if I click anywhere else on the page, I no longer see which category I just selected
* for any destructive/permanent action, like deletion, it's always great to have a confirmation
* when adding single empty space to the search input, content updates. I'd only use non-empty strings as valid search queries
* when opening comments modal, there's both "X" and "Close comments" buttons that do the same action. Feels a bit redundant
* when there're few elements on the page, ex. two cards, they are shown too far away from each other. I think it would look better if cards stayed in a grid with equal gap size between them, either center- or left-aligned relative to the container

### Code
* it's a convention to start boolean variable names with "is". In this case the setter starts with "is", which might be a bit confusing for whoever read your code: https://github.com/Oluwatosin12-bit/Kudos-Board/blob/ed42f66ef48727631b72059d56ef833e5720e1cb/FrontEnd/kudosBoard/src/BoardsPage.jsx#L12
I'd suggest the following naming instead:
```
 const [isRecentBoards, setIsRecentBoards] = useState(false) 
```
- I think one of these or both trigger an infinite look I mentioned above:
https://github.com/Oluwatosin12-bit/Kudos-Board/blob/ed42f66ef48727631b72059d56ef833e5720e1cb/FrontEnd/kudosBoard/src/BoardsPage.jsx#L30-L32
https://github.com/Oluwatosin12-bit/Kudos-Board/blob/ed42f66ef48727631b72059d56ef833e5720e1cb/FrontEnd/kudosBoard/src/CardsPage.jsx#L49-L51
I suggest you look into it to understand why and think how it can be fixed, to avoid in the future. In general, I suggest to keep an eye on the "Network" tabs in the browser developer tools to ensure no redundant requests are being sent.

- I've mentioned it previously, remember to do **explicit** null check for non-boolean values:
https://github.com/Oluwatosin12-bit/Kudos-Board/blob/ed42f66ef48727631b72059d56ef833e5720e1cb/FrontEnd/kudosBoard/src/BoardsPage.jsx#L37
https://github.com/Oluwatosin12-bit/Kudos-Board/blob/ed42f66ef48727631b72059d56ef833e5720e1cb/FrontEnd/kudosBoard/src/BoardsPage.jsx#L41
And in this case explicit check isn't really needed, cause this value is always a boolean:
https://github.com/Oluwatosin12-bit/Kudos-Board/blob/ed42f66ef48727631b72059d56ef833e5720e1cb/FrontEnd/kudosBoard/src/BoardsPage.jsx#L47
Can be just `if (recentBoards)`, or even better `if (isRecentBoards)`, if you renamed it following the naming convention mentioned above

- avoid repeating code where possible:
https://github.com/Oluwatosin12-bit/Kudos-Board/blob/ed42f66ef48727631b72059d56ef833e5720e1cb/FrontEnd/kudosBoard/src/BoardsPage.jsx#L47-L52
This could be written like so:
```
      if (recentBoards) {
        data.sort((a, b) => (b.id - a.id))
      }
      setBoards(data);
```
- I noticed that you used `BoardOptions` in multiple files, duplicating that code. 
Ideally, define it in one file, like `categories.js` and import from that file everywhere you need it.

A set of constant values can be defined as enum. JS doesn't support them natively, like TypeScript, but here's some documentation about how to define them - [link](https://www.geeksforgeeks.org/enums-in-javascript/)
In your case, you could create an object:
```
const CATEGORY = Object.freeze({
  ALL: Symbol('All'),
  RECENT: Symbol('Recent'),
  ...
})
```
and then use it to reference categories, for example here:
https://github.com/Oluwatosin12-bit/Kudos-Board/blob/ed42f66ef48727631b72059d56ef833e5720e1cb/FrontEnd/kudosBoard/src/BoardsPage.jsx#L60
you would have
```
if (category === CATEGORY.RECENT)
``` 
Unlike hardcoded string "Recent" this automatically ensures the correct value is used, easily avoiding typos.
Also, if you need to change a category name, you'd only need to do it in one place.

- this block of code could be inlined in JSX using `{}`; since it's only used once, no need to save as a variable
https://github.com/Oluwatosin12-bit/Kudos-Board/blob/ed42f66ef48727631b72059d56ef833e5720e1cb/FrontEnd/kudosBoard/src/BoardsPage.jsx#L86-L100

You could write it like so:
```
<div className="boardDisplay">
      {
         boards.map(board => <BoardDisplay key={board.id} {...board} />)
      }
</div>
```
Notice, that since all the props names of `BoardDisplay` match exactly the properties of the `board` object, you can just spread the object instead of passing each prop individually

- I saw a few instances in your code where you use a fragment (<></>) with a single JSX element in it. It should only be used when a component needs to return multiple elements, to create a wrapper:
https://github.com/Oluwatosin12-bit/Kudos-Board/blob/ed42f66ef48727631b72059d56ef833e5720e1cb/FrontEnd/kudosBoard/src/BoardCategories.jsx#L12-L23

- nit: no need for the spaces around "=": `className="searchArea"`:
https://github.com/Oluwatosin12-bit/Kudos-Board/blob/ed42f66ef48727631b72059d56ef833e5720e1cb/FrontEnd/kudosBoard/src/SearchBoards.jsx#L6

- in this case `id` and `label` values are the same, but in general, `label` should be used in the UI (visible to user), and `id` for referencing value in code:
https://github.com/Oluwatosin12-bit/Kudos-Board/blob/ed42f66ef48727631b72059d56ef833e5720e1cb/FrontEnd/kudosBoard/src/BoardCategories.jsx#L14-L21

So you'd use them like this 
```
<button
  onClick={() =>handleCategoryClick(id)}
   key={id}
   className="boardCategoryButton">
  {label}
</button>
```
-
